### PR TITLE
[Tested] OC-1376 add ValidatorTest for OCEL expression validator

### PR DESCRIPTION
**Pull request by NarzullaevObidjon**
## What

Add unit tests for `Validator` (OCEL expression validator).

- `ValidatorTest` — covers valid spacing variants, unbalanced parentheses, invalid tokens, unsupported operands, invalid operand pairs, and missing operands; each group uses `@ParameterizedTest` + `@ValueSource`

## Why
Part of `OC-1376`

## How to test
```bash
./gradlew test --tests "*.ValidatorTest"
```

## Checklist
- [x] Self-reviewed the diff
- [x] `./gradlew test` passes locally

### CODE REVIEW

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+@DisplayName("OCEL Expression Validator — unit")
+class ValidatorTest {
+
+    /** Postfix Validator*/
+    private final Validator validator = Validator.withCustomShallowEvaluator(ShallowEvaluatorType.POSTFIX);
+
+    // ── valid expressions ─────────────────────────────────────────────────────
+
+    @ParameterizedTest
+    @ValueSource(strings = {
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: **blocking**: The "valid expressions" group only tests spacing variants of a single expression `(#{...} || 1 = 3)`. That's one test case repeated 17 times, not coverage of the happy path. Add a separate test with valid cases for each comparison operator `(=, !=, >, <, >=, <=)`, each logical operator `(&&, ||, !)`, the custom keywords (`IsTypeOf`, `Matches`, `Like` and etc...), and parenthesized expressions. Without this, large parts of the validator's accept-path are untested.

**blocking**: No coverage for malformed reference IDs: `#{%%}`,` #{%abc (unclosed)`, `#{abc%} ``(missing leading %)`,` #{%not-hex%}`. The reference syntax is core to the validator and currently only the well-formed case is tested.

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Unit tests for {@link Validator}.
+ *
+ * No Spring context is loaded. The validator is instantiated directly.
+ * Run with: ./gradlew test
+ */
+@DisplayName("OCEL Expression Validator — unit")
+class ValidatorTest {
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:36]: **blocking**: Missing tests for null, empty string "", and whitespace-only "   " input. These are the most common inputs that crash validators in production. Use @NullSource and @EmptySource.

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            "(1 = \"asda\"",
+            "(1 = \"asda\")(",
+            "((1 = \"asda\")",
+            "1 = \"asda\")",
+            "((1 = \"asda\"))((",
+            ")1 = \"asda\"(",
+            ")",
+            "("
+    })
+    void validateThrowsInvalidParenthesesWhenParenthesesAreUnbalanced(String expression) {
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: **suggestion**: Extract the repeated "throws + check error code" pattern into a helper. The cast in `.satisfies(e -> assertThat(((InvalidExpressionException) e).getErrorCode())...)` appears four times and isInstanceOfSatisfying removes it cleanly:

`private void assertInvalid(String expression, ErrorCode expected) {
    assertThatThrownBy(() -> validator.validate(expression))
        .isInstanceOfSatisfying(InvalidExpressionException.class,
            e -> assertThat(e.getErrorCode()).isEqualTo(expected));
}`

Reasoning: removes duplication across five methods and eliminates the manual cast.

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        assertThatThrownBy(() -> validator.validate(expression))
+                .isInstanceOf(InvalidExpressionException.class)
+                .satisfies(e -> assertThat(((InvalidExpressionException) e).getErrorCode())
+                        .isEqualTo(ErrorCode.INVALID_PARENTHESES));
+    }
+
+    // ── invalid token ─────────────────────────────────────────────────────────
+
+    @ParameterizedTest
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: nit: Use a custom name on `@ParameterizedTes`t so failures show the input rather than [3]: `@ParameterizedTest(name = "[{index}] \"{0}\"")`.
Consider to apply for each `@ParameterizedTest`

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+/**
+ * Unit tests for {@link Validator}.
+ *
+ * No Spring context is loaded. The validator is instantiated directly.
+ * Run with: ./gradlew test
+ */
+@DisplayName("OCEL Expression Validator — unit")
+class ValidatorTest {
+
+    /** Postfix Validator*/
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: **nit**: The Javadoc on the validator field says `"Postfix Validator"` as a fragment. Either expand it ("Validator configured with the postfix shallow evaluator") or remove it.

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    private final Validator validator = Validator.withCustomShallowEvaluator(ShallowEvaluatorType.POSTFIX);
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: **question**: Is `Validator` stateless? The shared final instance assumes it is. If validate() mutates internal state, each test should get a fresh instance via `@BeforeEach`.

**question**: `ShallowEvaluatorType.POSTFIX` suggests other evaluator types exist. Should they have parallel test coverage, or is `POSTFIX` the only one in production use?

**Review by bobmustaev — CHANGES_REQUESTED** [review]

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            "#{%60b8d4c8f1d4a53d3c2f84d1%}   ||   1   =3"
+    })
+    void validateDoesNotThrowWhenExpressionHasVariousSpacing(String expression) {
+        assertThatCode(() -> validator.validate(expression)).doesNotThrowAnyException();
+    }
+
+    // ── invalid parentheses ───────────────────────────────────────────────────
+
+    @ParameterizedTest
+    @ValueSource(strings = {
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: suggestion: would be to great to apply following style adding comments for more clarity.

```
// 1. Integrity of the Reference Tag (The #{%id%} syntax)
@ParameterizedTest
@CsvSource({
    "'#{%}',             INVALID_TOKEN_FOUND",  // Empty ID
    "'#{%validID',       INVALID_TOKEN_FOUND",  // Unclosed tag
    "'{%validID%}',      INVALID_TOKEN_FOUND",  // Missing #
    "'#{%id1%} #{%id2%}', INSUFFICIENT_OPERAND"  // Two refs without an operator
})
void validateReferenceTagIntegrity(String expression, ErrorCode expected) {
    assertErrorCode(expression, expected);
}
```

**Review by bobmustaev — CHANGES_REQUESTED** [review]

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 2 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:36]: Added these test cases

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 5 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: I expanded it

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 6 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: 1. Yes, it is stateless
2. No, the shallow evaluator adds on top of basic syntax validation: operand-type compatibility and valid operator-operand pairings without evaulating expression totally. It works as second-level(deeper) checks.

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 7 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: good suggestion, i did it

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 4 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: ok, did it

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 3 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: ok, did it

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 1 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java]: this test case is for checking only different space variations. It is not supposed to check other cases.

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+                "(1 = 1) && (2 = 2)",
+        })
+        void validateDoesNotThrowForLogicalOperators(String expression) {
+            assertThatCode(() -> validator.validate(expression)).doesNotThrowAnyException();
+        }
+
+        // ── valid: custom keywords ────────────────────────────────────────────
+
+        @ParameterizedTest(name = "[{index}] \"{0}\"")
+        @ValueSource(strings = {
```
Comment (bobmustaev) [inline src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:93]: question: do we have other validation for other cases:
1. `"#{%60b8d4c8f1d4a53d3c2f84d1%} Contains '"#{%60b8d4c8f1d4a53d3c2f84d2%}',` // when expression has two references.
2. `#{%ID%} Like '%abc'; #{%ID%} Like 'abc%', #{%ID%} Like 'abc'`

**Review by bobmustaev — CHANGES_REQUESTED** [review]

File: src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java (hunk 8 above)
Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:93]: Yes, we should have these tests here as well. But all these tests are migrated from old test and no additional tests are not added. However, all other complex tests are written on PostfixExpressionProcessorTest. we can populate it with other tests that cover full functionality of the validator later

Comment (bobmustaev) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:93]: I don't want to postpone this  issue to later time. Would be great if you cover all cases in this unit test pls

Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:93]: Ok, i will do it

Comment (NarzullaevObidjon) [reply src/backend/src/test/java/com/becon/opencelium/backend/unit/ocel/ValidatorTest.java:93]: Added new tests to cover other untested cases
